### PR TITLE
Add rhel 8 cis

### PR DESCRIPTION
**Pull request by matttrach**

### CODE REVIEW

File: examples/os/rhel8cis/test.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+echo "I can exec!"
```
[the remark this answers is not in the document]
Comment (matttrach) [reply examples/os/rhel8cis/test.sh:3]: good catch!

File: variables.tf [hunk 2]
```diff
@@ -64,6 +64,17 @@ variable "image_admin_group" {
   default     = ""
 }
 
+variable "image_workfolder" {
+  type        = string
+  description = <<-EOT
+    The folder where scripts will be copied to and run from on the AMI.
+    This defaults to "/home/<initial_user>", and is usually safe.
+    If you are using an AMI where your home directory is mounted with noexec, you will need to change this.
```
[the remark this answers is not in the document]
Comment (matttrach) [reply variables.tf:72]: This sets the location where Terraform will place the script it runs to execute your script, basically a kind of chicken/egg situation. Unfortunately, if you can't exec where the Terraform script goes you can't see if the directory is mounted with noexec. We could check for that failure and adjust, but then we need to know what directory will allow exec and we would have to just keep guessing. My solution is to have a short list of images where we know the exec locations and let users select from those. If they want to use a custom AMI or some other AMI they will need to supply the expected exec location.

File: modules/server/types.tf [hunk 3]
```diff
@@ -1,32 +1,32 @@
 # storage sizes in GB, using gp3 storage type
 locals {
   types = {
-    small = {
-      id      = "t3.small",
+    small = { # minimum required for rke2 control plane node, handles 0-225 agents
```
[the remark this answers is not in the document]
Comment (matttrach) [reply modules/server/types.tf:4]: yeah, the update was to t3.medium from t3.small, based on the stress test info that just came out.

File: examples/os/rhel8cis/main.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  ssh_key_name        = local.key_name
+}
+
+# aws_access returns a security group object from the aws api, but the name attribute isn't the same as the Name tag
+# this is an rare example of when the name attribute is different than the Name tag
+module "TestRhel8Cis" {
+  depends_on = [
+    module.aws_access,
+  ]
+  source              = "../../../"
```
[the remark this answers is not in the document]
Comment (matttrach) [reply examples/os/rhel8cis/main.tf]: yes, good point, I will add that comment to all the tests.

File: variables.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (matttrach) [reply variables.tf:72]: It defaults to "/home/<initial user>" since usually the user is able to execute things in their own directory.

**Review by VestigeJ — APPROVED** [review]
LGTM

**Pull request merged**